File: Matura/pos-survival-kit-shareable/example-project/06_Janssen_animalShelterServer/src/main/java/janssen/spring/entities/AnimalShelter.java
package janssen.spring.entities;

import janssen.spring.JsonViews;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;import com.fasterxml.jackson.annotation.JsonView;
import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.*;
import java.util.Set;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
public class AnimalShelter {
  
  @Id
  @NotBlank
  @JsonView(JsonViews.Summary.class)
  protected String address;
  
  @OneToMany
  @ToString.Exclude
  protected Set<Animal> animals;
  
  @NotNull
  @Column(nullable = false)
  @JsonView(JsonViews.Summary.class)
  protected String phoneNumber;
  
  @OneToMany(mappedBy = "shelter")
  @ToString.Exclude
  @JsonIgnoreProperties({ "shelter" })
  protected Set<Sales> sales;
  
}

File: Matura/pos-survival-kit-shareable/example-project/06_Janssen_animalShelterApp/app/src/main/java/janssen/android/models/AnimalShelter.kt
package janssen.android.models

class AnimalShelter {
    
    var address: String = "?"
    var animals: List<Animal> = emptyList()
    var phoneNumber: String? = null
    var sales: List<Sales> = emptyList()
    
    override fun toString(): String {
        return "AnimalShelter(address=$address)"
    }
    
}
